docs/en/guides/pipeline-integrations/development/sgtk-developer-generating-path-and-publish.md

```diff
@@ -0,0 +1,567 @@
+
+When running your code in an environment where Shotgun is already started you can import the API by simply writing:
+
+```python
+import sgtk
+``` 
+
+If you are wanting to use the API outside of a Shotgun integration, for example, you might wish to test it in 
+your favorite IDE, then you will need to set the path to the API first:
+
+```python
+import sys
+sys.path.append("/shotgun/configs/my_project_config/install/core/python")
+
+import sgtk
+```
+
+If you're using distributed configs and your wanting to import `sgtk` in an environment where Toolkit hasn't already been bootstrapped, 
+you will need to take a different approach, please see the bootstrapping guide for more details.
+
+Now that you've imported the sgtk API you're ready to start using it.
+
+## Part 2 - Getting an Sgtk instance
+
+Now you've imported the `sgtk` API the next step towards the end goal of being able to generate a path and publish it, is
+to create an `Sgtk` instance.
+
+[`Sgtk`](https://developer.shotgunsoftware.com/tk-core/core.html#sgtk) is a class in the `sgtk` package that acts
+as the main interface to the API. Once we create an instance of `Sgtk` you will be able to do things like get a context,
+create folders, or access the templates.
+
+As the API documentation states, you don't create an instance of `Sgtk` directly. Here are some options for getting an `Sgtk` instance.
+
+1. You can get a `Sgtk` instance from the current engine if you are running the Python code within an environment where
+ the Shotgun integrations are already started. For example in Maya you could run the following:
+ 
+    ```python
+    # Get the engine that is currently running.
+    current_engine = sgtk.platform.current_engine()
+    
+    # Grab the already created Sgtk instance from the current engine.
+    tk = current_engine.sgtk
+    ```
+
+    You can access the `Sgtk` instance through the [`engine.sgtk`](https://developer.shotgunsoftware.com/tk-core/platform.html#sgtk.platform.Engine.sgtk) parameter.
+    The `engine.sgtk` should not be confused with or considered the same as the `sgtk` package that you imported in part 1. 
+    In this guide, it will be stored in a variable called `tk`.
+    If you're using the Shotgun Python console then you will have a pre-defined variable called `tk` which you can use.
+
+2. [`sgtk.sgtk_from_entity`](https://developer.shotgunsoftware.com/tk-core/initializing.html#sgtk.sgtk_from_entity) - 
+    if you are running in an environment where an engine hasn't already been started you can use this method to get an `Sgtk` instance based upon an entity id.
+    The entity whose id you are supplying must belong to the same project that the `sgtk` API was imported from. 
+ 
+3. [`sgtk.sgtk_from_path`](https://developer.shotgunsoftware.com/tk-core/initializing.html#sgtk.sgtk_from_path) -
+    much like the `sgtk_from_entity` except this will accept a path to a configuration or a path inside the project root.
```

Comment:
Changed to:
> ... except this will accept a path to a configuration or a path to or inside the project root folder. For example, a work file or shot folder.